cmd/cmd.go

```diff
@@ -1190,7 +1199,8 @@ Environment Variables:
 		RunE:    PullHandler,
 	}
 
-	pullCmd.Flags().Bool("insecure", false, "Use an insecure registry")
+	pullCmd.Flags().Bool("insecure", false, "Do not verify registry TLS")
+	pullCmd.Flags().Bool("http", false, "Use an http registry instead of https")
```

Comment:
We should look at the url in the arg for this instead of a new flag

Reply:
What URL?

Reply:
@jmorganca I disagree that what you describe should be the behavior. That is surprising behavior. I've never seen a type of insecure flag mean "downgrade the insecurity by 1 unit from whatever it is now".

Being more explicit would be easier to grok.

Reply:
Also, if there exists a URL already, then just use the scheme in the URL. Why even have the insecure flag? We could introduce a `-skip-verify` flag, but then we should probably think through the behavior when it is combined with a url that is http only. I would assume it fails and says "remove flag to proceed" in so many words.